backend/pkg/contenttype/middleware.go

```diff
@@ -0,0 +1,45 @@
+package contenttype
+
+import (
+	"errors"
+	"mime"
+	"net/http"
+	"slices"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/mendersoftware/mender-server/pkg/rest.utils"
+)
+
+func CheckJSON() gin.HandlerFunc {
+	return Middleware(false, "application/json")
+}
+
+func Middleware(checkCharest bool, contentTypes ...string) gin.HandlerFunc {
+
+	return func(c *gin.Context) {
+		contentType := c.GetHeader("Content-Type")
+		if !checkCharest {
+			contentType, _, _ = mime.ParseMediaType(contentType)
+		}
+
+		if c.Request.ContentLength > 0 && !slices.Contains(contentTypes, contentType) {
+			var errMsg string
+
+			if len(contentTypes) == 0 {
+				errMsg = "Bad Content-Type, expected no Content-Type"
+			} else {
+				for i, ctype := range contentTypes {
+					contentTypes[i] = "'" + ctype + "'"
+				}
```

Comment:
This is modifying the state of the function. The next time this middleware is called it will compare the quoted content type, and then the quoted quoted content type and so on...
Since the error message never changes, maybe we could build the error message once in the parent function (generator) scope?